app/assets/javascripts/discourse/app/components/related-messages.gjs

```diff
@@ -0,0 +1,95 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import BasicTopicList from "discourse/components/basic-topic-list";
+import concatClass from "discourse/helpers/concat-class";
+import i18n from "discourse-common/helpers/i18n";
+import getURL from "discourse-common/lib/get-url";
+import I18n from "discourse-i18n";
+
+const LIST_ID = "related-Messages";
+
+export default class RelatedMessages extends Component {
+  @service moreTopicsPreferenceTracking;
+  @service currentUser;
+
+  get hidden() {
+    return this.moreTopicsPreferenceTracking.get("selectedTab") !== LIST_ID;
+  }
+
+  @action
+  registerList() {
+    this.moreTopicsPreferenceTracking.registerTopicList({
+      name: I18n.t("related_messages.pill"),
+      id: LIST_ID,
+    });
+  }
+
+  @action
+  removeList() {
+    this.moreTopicsPreferenceTracking.removeTopicList(LIST_ID);
+  }
+
+  get targetUser() {
```

Comment:
a micro-optimization but I'd pop a `@cached` on this bad boy 😅 it's used three times below, so all those checks and iterating is repeated again and again